node/node-sdk.md

```diff
@@ -0,0 +1,187 @@
+# NodeJS SDK Proposal
+
+NodeJS is a perfect candidate for authoring Temporal workflows, by leveraging V8 isolates for running workflow code,
+we can create a JS runtime that's completely deterministic.
+
+The SDK steers developers to write their workflows and activities in TypeScript but vanilla JS is also supported.
+All examples in this proposal are written in TypeScript.
+
+## Table Of Contents
+- [Activities](#activities)
+    - [Examples](#examples)
+    - [Registration](#registration)
+    - [Payloads](#payloads)
+- [Workflows](#workflows)
+    - [Example](#example)
+    - [Activity Options](#activity-options)
+    - [Activity Type Declarations](#activity-type-declarations)
+    - [Registration](#registration-1)
+    - [Limitations](#limitations)
+    - [Signals](#signals)
+    - [Queries](#queries)
+    - [Logging](#logging)
+    - [Sessions](#sessions)
+    - [Deterministic Time](#deterministic-time)
+    - [Error Handling](#error-handling)
+    - [Cancellation](#cancellation)
+    - [Long Histories](#long-histories)
+
+
+## Activities
+Activity implementations are just JS functions, they run on the default NodeJS runtime, along with the Temporal SDK, and can import arbitrary npm packages.
```

Comment:
I don't know JS, but in Go and Java we encourage use of object methods for activities. This way any external dependencies can be injected into the object fields during the worker initialization phase. Is it antipattern in JS?

Reply:
Yes, normally you'd have a function, which would take a context argument if something needs to be injected.

Reply:
i dont know about "normally", but es6 objects and classes handle this fine